### PR TITLE
feat: Add password confirmation

### PR DESCRIPTION
**Pull request by casperklein**
# Description

Fixes: #4071 (typo when entering the password)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [ ] If necessary, I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] **I have added information about changes made in this PR to `CHANGELOG.md`**

### CODE REVIEW

**Comment by casperklein** [comment]
@georglauterbach @polarathene 
See my second commit https://github.com/docker-mailserver/docker-mailserver/pull/4072/commits/7a4186fada90dd2cf20ff6bb765f1afae4b831c8. Any idea why this was implemented this way in the first place?

If you agree, I'll change this logic in the other files as well.

**Review by polarathene — APPROVED** [review]

File: target/scripts/helpers/database/manage/postfix-accounts.sh [hunk 1]
```diff
@@ -98,9 +98,14 @@ function __account_already_exists() {
 
 # Also used by addsaslpassword
 function _password_request_if_missing() {
+  local CONFIRM
```
Comment (polarathene) [inline target/scripts/helpers/database/manage/postfix-accounts.sh:101]: Perhaps `PASSWD_CONFIRM` or similar would be a bit better in context. I'm not fussed either way though.

**Comment by georglauterbach** [comment]
I don't know either why we use `${*}`, `${2}` seems more appropriate 👍🏼

**Review by georglauterbach — DISMISSED** [review]

**Review by polarathene — APPROVED** [review]
LGTM 👍

**Review by georglauterbach — APPROVED** [review]

**Pull request merged**